queries.sql

```diff
@@ -15,3 +15,52 @@ SELECT * FROM animals WHERE neutered = 'true';
 SELECT * FROM animals WHERE name <> 'Gabumon';
 
 SELECT * FROM animals WHERE weight_kg >= 10.4 AND weight_kg <= 17.3;
+
+-- TRANSACTIONS
+
+BEGIN;
+UPDATE animals SET species = 'digimon' WHERE name LIKE '%mon';
+UPDATE animals SET species = 'pokemon' WHERE species IS NULL;
+SELECT * FROM animals;
+COMMIT;
+
+BEGIN;
+DELETE FROM animals;
+ROLLBACK;
+SELECT * FROM animals;
+
+BEGIN;
+DELETE FROM animals WHERE date_of_birth > '2022-01-01';
+COMMIT;
+SELECT * FROM animals;
+
+
+BEGIN;
+SAVEPOINT bmg1;
+UPDATE animals SET weight_kg = weight_kg * -1;
+SELECT * FROM animals;
+ROLLBACK;
+SELECT * FROM animals;
+
+BEGIN;
+UPDATE animals SET weight_kg = weight_kg * -1 WHERE weight_kg < 0;
+COMMIT;
+SELECT * FROM animals;
```

Comment:
You've made good progress in implementing the SQL transactions according to the project requirements. Your code structure is generally correct, but there are a few points to consider:

1. **Savepoint Usage:** You've correctly used a savepoint (`SAVEPOINT bmg1;`) and then attempted to rollback to that savepoint (`ROLLBACK;`). However, it appears that you've used a generic `ROLLBACK;` statement without specifying the savepoint. To achieve the desired behavior of rolling back to the savepoint, use `ROLLBACK TO bmg1;` instead.

Here's an updated version of your code with these suggestions applied:

```sql
-- Transaction 1: Delete and Rollback
BEGIN;
DELETE FROM animals WHERE date_of_birth > '2022/01/01';
SAVEPOINT bmg1;
UPDATE animals SET weight_kg = weight_kg * -1;
ROLLBACK TO bmg1;
COMMIT;
SELECT * FROM animals;

-- Transaction 2: Update Negative Weights and Commit
BEGIN;
UPDATE animals SET weight_kg = weight_kg * -1 WHERE weight_kg < 0;
COMMIT;
SELECT * FROM animals;
```

Your code structure is on the right track, and you've effectively followed the project requirements. Just ensure that you use `ROLLBACK TO` with the specific savepoint name to achieve the desired rollback behavior.

Keep up the good work, and make sure to continue practicing SQL transactions to maintain data consistency and integrity.

Best regards,
Temitope
Code Reviewer
01-09-2023